### PR TITLE
fix: remove bug that make users insert wrong isbn in market

### PR DESCRIPTION
**Pull request by ellegint**
Even though we'll change db to sqlite files, to launch the first version of the bot this fix is very important.
In particular:

- New checks are added to the `sell.py` module, in order to avoid users to put something strange.
- ISBN entries are now checked correctly inside the book database.

### CODE REVIEW

File: module/sell.py [hunk 1]
```diff
@@ -13,47 +13,50 @@ def sell(update: Update, context: CallbackContext) -> None:
...
-                url="https://catalogo.unict.it/search/i?SEARCH=" + user_isbn + "&sortdropdown=-&searchscope=9"
-                x = requests.get(url)
-                soup = BeautifulSoup(x.content, 'html.parser')
-                check = str(soup.findAll('td'))
-                if "No matches found" not in check:
-                    title_and_author  = soup.find('strong').text
+        if len(user_isbn) == 10 or len(user_isbn) == 13:
+            try:
+                format(float(message.split()[2].replace(',', '.')), '.2f')
+                prezzo = str(format(float(message.split()[2].replace(',', '.')), '.2f'))
```
Comment (Helias) [inline module/sell.py]:
```suggestion
                price = str(format(float(message.split()[2].replace(',', '.')), '.2f'))
```

File: module/sell.py [hunk 2]
```diff
@@ -13,47 +13,50 @@ def sell(update: Update, context: CallbackContext) -> None:
...
+                username ="@"+str(context.bot.get_chat(chat_id)['username'])
```
Comment (Helias) [inline module/sell.py]: and, if the username is not present?

File: module/sell.py [hunk 3]
```diff
@@ -13,47 +13,50 @@ def sell(update: Update, context: CallbackContext) -> None:
...
+                        df = get_on_sale()
+                        new_row = pd.DataFrame({'ISBN':str(isbn),'Titolo':str(title), 'Autori':str(authors), 'Venditore':str(username), 'Prezzo':str(prezzo)}, index=[len(df)])
+                        df = pd.concat([df, new_row])
+                        df.to_csv(SHOP_DB_PATH)
+                        new_book = pd.DataFrame({'ISBN':str(isbn),'Titolo':str(title), 'Autori':str(authors)}, index=[len(books)])
+                        books = pd.concat([books, new_book])
+                        books.to_csv(BOOKS_DB_PATH)
+                        context.bot.send_message(chat_id, "Il libro è stato aggiunto al database.")
+                    else:
+                        context.bot.send_message(chat_id, "Libro non trovato. Controlla di aver inserito correttamente l'ISBN.")
```
Comment (Helias) [inline module/sell.py]: what about putting all this of the below `else` in a separate function like `is_book_present_in_unict()`?
The function could return a boolean value and a possible result could be
```python
                    if is_book_present_in_unict():
                        context.bot.send_message(chat_id, "Il libro è stato aggiunto al database.")
                    else:
                        context.bot.send_message(chat_id, "Libro non trovato. Controlla di aver inserito correttamente l'ISBN.")
```

File: module/sell.py (hunk 2 above)
Comment (ellegint) [reply module/sell.py]: Check added!

File: module/sell.py (hunk 3 above)
Comment (ellegint) [reply module/sell.py]: Refactor done!

File: module/sell.py (hunk 1 above)
Comment (ellegint) [reply module/sell.py]: Forgot about that. Fixed!

File: module/sell.py [hunk 4]
```diff
@@ -1,62 +1,57 @@
...
                         context.bot.send_message(chat_id, "Il libro è stato aggiunto al database.")
-                    else:
-                        context.bot.send_message(chat_id, "Libro non trovato. Controlla di aver inserito correttamente l'ISBN.")
-            except Exception as e:
-                print(str(e))
-                context.bot.send_message(chat_id, "Prezzo non valido.")
+                        return
+                    
+                    context.bot.send_message(chat_id, "Libro non trovato. Controlla di aver inserito correttamente l'ISBN.")
+                    return
```
Comment (Helias) [inline module/sell.py]: I don't think that this is important, when a function end by default return None

Comment (ellegint) [reply module/sell.py]: Right, I forgot to remove that

**Review by Helias — APPROVED** [review]

**Comment by Helias** [comment]
Good job!
Last thing that I was missing, you could add the settings.yaml file to the `.gitignore`

**Comment by Helias** [comment]
wait, I checked it now

image

No, you should not change it.

Let me explain in another way:
- you create in your project settings.yaml.dist
- a developer clone your project and he wants to use it, so he needs to change settings.yaml.dist
- but, if the developer does this, git will track the settings.yaml.dist and this is not really good, by mistake the developer could push his token to a repository => so, a good practice is copying the settings.yaml.dist into settings.yaml keeping settings.yaml in `.gitignore` and settings.yaml.dist for development purpose

Let change the name back and just add the settings.yaml into the .gitignore, moreover, you could explain this mechanism into the README.md like we do for DMI/Spotted bot

**Pull request merged**